core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -112,9 +112,19 @@ class LogCleaner(initialConfig: CleanerConfig,
 
   private[log] val cleaners = mutable.ArrayBuffer[CleanerThread]()
 
+  /**
+   * scala 2.12 does not support maxOption so we handle the empty manually.
+   * @param f to compute the result
+   * @return the computed value or 0 if there is no cleaner
+   */
+  private def computeCleanerThread(f: Seq[CleanerThread] => Int): Int =
```

Comment:
a workaround for scala 2.12

Reply:
Maybe we could turn this into a `foldLeft` and avoid the `map`. For example:
```scala
  private def maxOverCleanerThreads(f: CleanerThread => Int): Int = {
    cleaners.foldLeft(0, (acc: Int, thread: CleanerThread) => math.max(acc, f(thread)))
  }
```